tests/virt_autotest/libvirt_isolated_virtual_network.pm

```diff
@@ -43,26 +43,20 @@ sub run_test {
     upload_logs "vnet_isolated.xml";
     assert_script_run("rm -rf vnet_isolated.xml");
 
-    my $gi_vnet_isolated;
+    my ($mac, $nic);
     foreach my $guest (keys %xen::guests) {
         record_info "$guest", "ISOLATED NETWORK for $guest";
         #figure out that used with virtio as the network device model during
         #attach-interface via virsh worked for all sles guest
-        assert_script_run("virsh attach-interface $guest network vnet_isolated --model virtio --live");
-        #Get the Guest IP Address from ISOLATED NETWORK
-        if (get_var("XEN") || check_var("HOST_HYPERVISOR", "xen")) {
-            my $mac_isolated = script_output("virsh domiflist $guest | grep vnet_isolated | grep -oE \"[[:xdigit:]]{2}(:[[:xdigit:]]{2}){5}\"");
-            script_retry "ip neigh | grep $mac_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 90, retry => 9, timeout => 90;
-            $gi_vnet_isolated = script_output("ip neigh | grep $mac_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        }
-        else {
-            script_retry "virsh net-dhcp-leases vnet_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"", delay => 90, retry => 9, timeout => 90;
-            $gi_vnet_isolated = script_output("virsh net-dhcp-leases vnet_isolated | grep -oE \"([0-9]{1,3}[\.]){3}[0-9]{1,3}\"");
-        }
+        $mac = '00:16:3e:32:' . (int(rand(89)) + 10) . ':' . (int(rand(89)) + 10);
+        assert_script_run("virsh attach-interface $guest network vnet_isolated --model virtio --mac $mac --live", 60);
+        $nic = script_output "ssh root\@$guest ls /sys/class/net/ | grep -v 'lo\\|eth0'";
+        assert_script_run("ssh root\@$guest ip a a 192.168.122.207/24 dev $nic");
         #Confirm ISOLATED NETWORK
-        assert_script_run("! ssh root\@$gi_vnet_isolated 'ping -c2 -W1 openqa.suse.de'");
+        assert_script_run("ping -c3 192.168.122.207");
+        assert_script_run("ssh root\@192.168.122.207 'curl -o /dev/null --insecure --interface 192.168.122.207 https://openqa.suse.de'", 60);
         save_screenshot;
-        assert_script_run("virsh detach-interface $guest network --current");
+        assert_script_run("virsh detach-interface $guest network --mac $mac --current");
```

Comment:
There should be confirmed guest system used with isolated private IP cannot reach any other machines on the LAN.

Reply:
I am now checking that the gateway is not forwarding the traffic. Everything is now under the `test_network_interface()` function in `lib/virt_autotest/virtual_network_utils.pm` file.